plugins/modules/ndo_interface_setting.py

```diff
@@ -0,0 +1,831 @@
+            speed=dict(type="str", choices=["100M", "1G", "10G", "25G", "40G", "50G", "100G", "200G", "400G", "inherit"]),
+            auto_negotiation=dict(type="str", choices=["on", "off", "on_enforce"]),
+            vlan_scope=dict(type="str", choices=["global", "port_local"]),
+            cdp_admin_state=dict(type="str", choices=["enabled", "disabled"]),
+            lldp=dict(
+                type="dict",
+                options=dict(
+                    status=dict(type="str", choices=["enabled", "disabled"]),
+                    transmit_state=dict(type="str", choices=["enabled", "disabled"]),
+                    receive_state=dict(type="str", choices=["enabled", "disabled"]),
+                ),
+            ),
+            domains=dict(type="list", elements="str"),
+            port_channel_mode=dict(type="str", choices=list(PORT_CHANNEL_MODE_MAP)),
+            min_links=dict(type="int"),
+            max_links=dict(type="int"),
+            control=dict(type="list", elements="str", choices=list(CONTROL_MAP)),
+            load_balance_hashing=dict(type="str", choices=list(LOAD_BALANCE_HASHING_MAP)),
+            sync_e=dict(type="str"),
+            link_level_debounce_interval=dict(type="int"),
+            link_level_bring_up_delay=dict(type="int"),
+            link_level_fec=dict(type="str", choices=list(LINK_LEVEL_FEC_MAP)),
+            l2_interface_qinq=dict(type="str", choices=list(L2_INTERFACE_QINQ_MAP)),
+            l2_interface_reflective_relay=dict(type="str", choices=["enabled", "disabled"]),
+            stp_bpdu_filter=dict(type="str", choices=["enabled", "disabled"]),
+            stp_bpdu_guard=dict(type="str", choices=["enabled", "disabled"]),
+            llfc_transmit_state=dict(type="str", choices=["enabled", "disabled"]),
+            llfc_receive_state=dict(type="str", choices=["enabled", "disabled"]),
+            mcp=dict(
+                type="dict",
+                options=dict(
+                    admin_state=dict(type="str", choices=["enabled", "disabled"]),
+                    strict_mode=dict(type="str", choices=["on", "off"], aliases=["mcp_mode"]),
+                    initial_delay_time=dict(type="int"),
+                    transmission_frequency_sec=dict(type="int"),
+                    transmission_frequency_msec=dict(type="int"),
+                    grace_period_sec=dict(type="int"),
+                    grace_period_msec=dict(type="int"),
+                ),
+            ),
+            pfc_admin_state=dict(type="str", choices=["on", "off", "auto"]),
+            access_macsec_policy=dict(type="str"),
+            state=dict(type="str", choices=["absent", "query", "present"], default="query"),
+        )
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "present", ["interface_policy_group", "interface_policy_group_uuid", "interface_type"], True],
```

Comment:
The `interface_type` is required during the object creation; I think you need to change the required_if condition.

Reply:
Added additional checks just in create case and added tasks to check the condition